base_model_restrict_update/models/ir_model_access.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models, tools
+from odoo.exceptions import AccessError
+from odoo.tools.translate import _
+
+
+class IrModelAccess(models.Model):
+    _inherit = "ir.model.access"
+
+    @api.model
+    @tools.ormcache_context(
+        "self._uid", "model", "mode", "raise_exception", keys=("lang",)
+    )
+    def check(self, model, mode="read", raise_exception=True):
+        if self.env.su:
+            return True
+        res = super().check(model, mode, raise_exception)
```

Comment:
```suggestion
        self.flush()
        res = super().check(model, mode, raise_exception)
```

Reply:
@Saran440 
I tested by removing rights on Emplyee, Contact and Expense. Now, I can replicate the problem.

I try with .fulsh() and still have the same error.